TypeScript <2.6 compatibility issue (solution known!)

## Issue

I recently upgraded rollup from `0.55.x` to `0.56.x` in my project (see https://github.com/dominique-mueller/angular-package-builder/pull/62), and it seems the new release broke compatibility with TypeScript `2.5` and lower (while TypeScript `2.6` and higher continue to work as expected).

This can be seen easily in the project's test matrix on Travis CI: https://travis-ci.org/dominique-mueller/angular-package-builder/builds/341990680.

The following error occurs:

``` text
node_modules/rollup/dist/typings/ast/CallOptions.d.ts(17,52): error TS1110: Type expected.
node_modules/rollup/dist/typings/ast/ExecutionPathOptions.d.ts(27,22): error TS1110: Type expected.
node_modules/rollup/dist/typings/ast/ExecutionPathOptions.d.ts(33,70): error TS1110: Type expected.
node_modules/rollup/dist/typings/ast/ExecutionPathOptions.d.ts(34,101): error TS1110: Type expected.
node_modules/rollup/dist/typings/ast/ExecutionPathOptions.d.ts(35,68): error TS1110: Type expected.
node_modules/rollup/dist/typings/ast/ExecutionPathOptions.d.ts(36,101): error TS1110: Type expected.
node_modules/rollup/dist/typings/ast/ExecutionPathOptions.d.ts(37,105): error TS1110: Type expected.
node_modules/rollup/dist/typings/ast/ExecutionPathOptions.d.ts(38,99): error TS1110: Type expected.
node_modules/rollup/dist/typings/ast/ExecutionPathOptions.d.ts(40,39): error TS1110: Type expected.
node_modules/rollup/dist/typings/ast/ExecutionPathOptions.d.ts(51,94): error TS1110: Type expected.
node_modules/rollup/dist/typings/ast/ExecutionPathOptions.d.ts(52,59): error TS1110: Type expected.
node_modules/rollup/dist/typings/ast/ExecutionPathOptions.d.ts(53,48): error TS1110: Type expected.
node_modules/rollup/dist/typings/ast/ExecutionPathOptions.d.ts(54,40): error TS1110: Type expected.
node_modules/rollup/dist/typings/ast/ExecutionPathOptions.d.ts(55,26): error TS1110: Type expected.
node_modules/rollup/dist/typings/ast/ExecutionPathOptions.d.ts(56,49): error TS1110: Type expected.
node_modules/rollup/dist/typings/ast/scopes/Scope.d.ts(39,45): error TS1110: Type expected.
node_modules/rollup/dist/typings/ast/scopes/Scope.d.ts(39,52): error TS1005: '=' expected.
```
<br>

## Research

The problem exists when a class which gets exported as `default` has a static method creating and returning an instance of this exact class. For instance, the following typing is published right now (first line of error message):

``` typescript
export default class CallOptions implements CallCreateOptions {
    static create(callOptions: CallCreateOptions): default;
}
```

The correct typing, however, should be:

``` typescript
export default class CallOptions implements CallCreateOptions {
    static create(callOptions: CallCreateOptions): CallOptions;
}
```

A bit of research showed that, between `0.55` and `0.56`, rollup internally upgraded TypeScript from version `2.6` to `2.7` for its own build process (see https://github.com/rollup/rollup/commit/bf3096f2533e3ad1a329054428bc89f4aabf949e), with version `2.7.1` getting used right now (see https://github.com/rollup/rollup/commit/bf3096f2533e3ad1a329054428bc89f4aabf949e#diff-32607347f8126e6534ebc7ebaec4853dR4330).

Further research showed that this exact issue is already known (see https://github.com/Microsoft/TypeScript/issues/21958) - and got fixed with TypeScript `2.7.2`.

<br>

## Solution

Upgrading TypeScript to `>= 2.7.2` and releasing a new patch version should do the fix.

--- a/package.json
+++ b/package.json
@@ -90,7 +90,7 @@
     "source-map": "^0.6.1",
     "source-map-support": "^0.5.3",
     "sourcemap-codec": "^1.4.0",
-    "typescript": "^2.7.1",
+    "typescript": "^2.7.2",
     "uglify-js": "^3.3.10"
   },
   "files": [

